### PR TITLE
[CSL-2343] Remove wallet/node in favour of wallet-new/server

### PR DESCRIPTION
**Pull request by akegalj**
https://iohk.myjetbrains.com/youtrack/issue/CSL-2343

Tested:

* [x] cardano-sl building with `scripts/build/cardano-sl.sh -O2 --no-asserts`
* [x] local cluster run with ` scripts/launch/demo-with-wallet-api.sh`

* [x] Daedalus acceptance tests https://github.com/input-output-hk/daedalus/tree/develop#testing passing:
```
$ npm run test

61 scenarios (61 passed)
450 steps (450 passed)
11m48.572s
```

* [x] manuall testing: creating two wallets, redeeming ADA, sending money from one wallet to another
look

* [x] CI build passes .- testing locally via https://github.com/input-output-hk/daedalus/tree/develop#cidev-build-scripts for linux
      - the results will be also visible on this PR for other OSes

### CODE REVIEW

**Comment by akegalj** [comment]
@darko-mijic @domenkozar would you mind checking am I missing something?

I have checked appveyor.yml and by my naive understanding it seems correct - but I am not 100% sure.

**Comment by akegalj** [comment]
> What about scripts/launch/demo-with-wallet-api? I think we should simply rename demo-with-legacy-new-wallet-api -> demo-with-wallet-api as that's what we'll be using from now on;

at the moment they are both the same - so I can just remove `demo-with-legacy-new-wallet-api.sh` . Didn't want to introduce any breaking changes atm. I will remove it then

> Have you grepped through the codebase to make sure our use of cardano-node is not breaking any pre-existing script?

I have - but my understanding is that new cardano wallet executable (new cardano-node with default settings) shouldn't break old cardano wallet executable (old cardano-node). I will run some of the scripts more to try to test it (I did grep it though) already

**Comment by akegalj** [comment]
@jmitchell Can you help me figure out failure I am getting from this build.

Basically what we are doing here is:
 * removing `cardano-node` executable from `wallet/*.cabal`
 * renaming `cardano-new` -> `cardano-node` executable from `wallet-new/*cabal`

With latest 3 commits: `"Fix CI*"` I have tried to fixed build and I made some progress but I am stuck now. I believe there is not much more to change, but I am not that familiar with build internals.

Can you please have a look at failure and/or review changes I have done to the build in this PR (latest 3 commits)

**Comment by jmitchell** [comment]
@akegalj, I pushed a change I suspect will help.

**Comment by akegalj** [comment]
@jmitchell thank you! That fixes the build

**Comment by akegalj** [comment]
@jmitchell would you be kind as well to have a look at the changes I have done to build files - for me to have a bit more confidence I am not doing something terribly wrong.

**Comment by akegalj** [comment]
@adinapoli-iohk I believe it should be ready to go. I can squash all the commits and force push (or we can do it when merging to `develop` - not sure which one you prefer more)

**Comment by adinapoli-iohk** [comment]
Looks good @akegalj , thanks! (and thanks @jmitchell for helping with the CI!).

Before merging, I would like to make sure this is on @nikolaglumac, @darko-mijic and @domenkozar 's radar.

Needless to say we should merge (my 2 cents, at least 😁) this one squashing it, so reverting is a no-brainer in case of problems.

File: scripts/ci/ci.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ if [[ ("$OS_NAME" == "linux") && ("$BUILDKITE_BRANCH" == "master") ]];
   else with_haddock=false
 fi
 
-targets="cardano-sl cardano-sl-auxx cardano-sl-tools cardano-sl-wallet"
+targets="cardano-sl cardano-sl-auxx cardano-sl-tools cardano-sl-wallet cardano-sl-wallet-new"
```
Comment (domenkozar) [inline scripts/ci/ci.sh]: we should delete `cardano-sl-wallet`

Comment (adinapoli-iohk) [reply scripts/ci/ci.sh]: @domenkozar If this is building the executable yes, but `cardano-wallet-new` has still a _library_ dependency on `cardano-wallet`, so maybe that's fine it keep it there for now 😉 (but you tell me, ofc).

**Pull request merged**